src/main/java/org/scijava/convert/StringToNumberConverter.java

```diff
@@ -51,6 +50,22 @@ public Class<String> getInputType() {
 	@Override
 	public boolean canConvert(Object src, Class<?> dest) {
 		if (!Types.isAssignable(src.getClass(), String.class)) return false;
-		return Types.isAssignable(Types.box(dest), Number.class);
+		// The only way to know if the conversion is valid is to actually do it.
+		try {
+			String srcString = (String) src;
+			sane(dest).getConstructor(String.class).newInstance(srcString);
```

Comment:
This reflection, while slick, concerns me w.r.t. the Java 11+ restrictions on reflection. But it's fine for now, we can always fix later as needed.

Reply:
Yes, you are right. `DefaultConverter` does the same thing, though, so I think you are right to fix these later.